### DISCUSSION TITLE
MissionControl 0.9.2

### DISCUSSION THREAD

**Discussion by ndeadly** [ndeadly/MissionControl#618] [General]
<p align="left">
<img alt="GitHub Releases" src="https://img.shields.io/github/downloads/ndeadly/MissionControl/v0.9.2/total">
<a href="https://discord.gg/gegfNZ5Ucz"><img alt="Discord Server" src="https://discordapp.com/api/guilds/905069757361971303/widget.png?style=shield"/></a>
</p>

This is the nineteenth official release of Mission Control.

This is release addresses a couple of critical issues introduced in the previous version.
- An issue caused by setting up a USB interface filter with the same parameters as the one used by the `hid` module. This caused the `hid` module to stop receiving USB connection events, breaking support for official and third-party USB wired controllers and dongles. Sorry guys!

- An issue where Mission Control would crash if the Bluetooth pairing database was full when trying to pair a Dualshock 3 controller.

New users landing on this release page should first check out the readme on the [main project page](https://github.com/ndeadly/MissionControl#readme) for the official project documentation. There you can find installation and usage instructions along with an FAQ section that will answer most of your questions.

### Changelog:
- Modified the usb interface filter used in Dualshock 3 pairing so as not to clash with the default filter used by the `hid` module
- Fixed an issue where the system would crash when attempting to pair a Dualshock 3 controller when the Bluetooth pairing database contains the maximum number of paired hid devices.

<hr /><em>This discussion was created from the release <a href='https://github.com/ndeadly/MissionControl/releases/tag/v0.9.2'>MissionControl 0.9.2</a>.</em>

**Comment by farmgram** [comment]
不知道xbox series最新固件能连接了不

**Reply by ndeadly** [reply]
English please. You can use a service like https://www.deepl.com/translator if you need to translate.

**Reply by farmgram** [reply]
xbox2020 latest firmware does not seem to support connection

**Reply by farmgram** [reply]
16.0.1&1.5.1

**Reply by ndeadly** [reply]
The new Xbox controllers aren't supported yet as they use a different type of Bluetooth connection. I'm working on adding support for it.

**Reply by farmgram** [reply]
I understand. Thank you very much.

**Reply by nznznz746** [reply]
encourage！wait for xbox2020

**Comment by zbcyd888** [comment]
Does it support 16.0.2?

**Reply by ndeadly** [reply]
Yes. Generally speaking, an update is only required if Atmosphere requires an update. Also, I will never put out a new release that doesn't have support for the latest firmware at that time.

**Comment by alwinmuhaimin** [comment]
Hi @ndeadly, when I plugged the usb cable to the dock (already in change grip/order) my ds3 always detect as usb controller. Then I take off the usb cable, all 4 led are flashing and I pressed PS button but nothing happened until the controller is turn off. Did I missed some steps?

**Reply by ndeadly** [reply]
Is it an official Sony DS3 or a clone? Some clones may work, but I make no guarantees as they are known to have quirks affecting compatibility. If it's not official, you may have some level of success with the build I posted [here](https://github.com/ndeadly/MissionControl/issues/615)

> my ds3 always detect as usb controller

Sounds like you've probably got sys-con installed too. I had someone test to make sure my changes didn't break sys-con, but I'm not sure if anybody checked that sys-con didn't break my USB code for DS3. It might be worth disabling/removing sys-con if you've got it and testing if that makes any difference.

**Reply by ndeadly** [reply]
Also keep in mind that the pairing doesn't seem to be 100% reliable. Sometimes I've had to try again once or twice for it to work.

**Comment by BerserkSoul1989** [comment]
Hi there :D
I have tried to pair my DS3 but no luck yet. The control shows the fast red LEDs but when I press PS button it keeps blinking but slower, like it's searching for some host.
I'm using a blue one, maybe it's generic

**Reply by ndeadly** [reply]
I only technically support the official Sony Dualshock 3 controller. While some of the generic ones may work, they are known to have all kinds of quirks that may affect compatibility. You could try the build I posted in [this issue](https://github.com/ndeadly/MissionControl/issues/615) and see if it will alllow your controller to connect. This may work with varying levels of success depending on the controller. Still not sure how to handle this officially though, as I don't really know how to reliably tell the generic controllers apart from the official ones.

**Comment by MonfGeiger** [comment]
i saw a previous comment that 0.9.2 supports 16.0.2, can i have any assurance that it will also work with 16.0.3?
no offense meant, but last time i updated my atmo setup and didnt update this particular piece of homebrew, my atmo setup refused to boot properly until i either removed missioncontrol, or updated it

**Reply by ndeadly** [reply]
Yes, it's compatible with 16.0.3. Typically, a micro version bump (increment in the last number of the firmware version) won't require a Mission Control update. I don't have much of a social media presence, but I do make announcements regarding version compatibility on my discord server if you wanted to keep up.

> no offense meant, but last time i updated my atmo setup and didnt update this particular piece of homebrew, my atmo setup refused to boot properly until i either removed missioncontrol, or updated it

None taken. This is normal and not something I can do anything about. libstratosphere (the Atmosphere library for developing sysmodules) checks the firmware against a list of known firmware versions and will abort if the current version indicates significant updates have been applied. Presumably this is to prevent homebrew from calling system services with incorrect arguments should anything have changed. The reason Mission Control has a "reputation" for needing to be removed or updated is simply because a) it's one of the few sysmodules that actually uses libstratosphere, and b) it's probably the most popular of those that do.

**Reply by Yunikonius** [reply]
Hi @ndeadly . I’ve updated switch to 16.0.3, and my xone controller doesn’t work now. Before update it worked, so i think it’s not because of BLE. Ps i reinstalled MissionControl, no effect.

**Reply by ndeadly** [reply]
@Yunikonius If you're absolutely certain you haven't updated to the BLE firmware, these are the usual suspects when a controller stops working https://github.com/ndeadly/MissionControl/discussions/618#discussioncomment-6195462

**Comment by Masamune3210** [comment]
Trust me, aborting on unknown state is a lot better than blindly calling into things and accidentally wrecking into something lol

**Reply by ndeadly** [reply]
I presume you meant to post this as a reply to the post above ;) (man the github UI for this is horrible, I've done the same before too)

**Reply by Masamune3210** [reply]
Yeaaah, it didn't even show up as separate in the GitHub app. Mobile GitHub is not a fun time lol

**Comment by MonfGeiger** [comment]
well, i hope this sysmodule lasts as long as it can, saves me the trouble of getting a new non-joycon controller and lets me reuse the ds4 controller i use for my PC (also gave said controller a higher capacity battery)

**Reply by ndeadly** [reply]
I don't think you've got anything to worry about. I have no intention of dropping the project soon, and there's enough demand there that I'm sure somebody else would pick it up if I stopped updating.

**Comment by DX12development** [comment]
i've got an issue where bluetooth audio connects but doesn't output anything while my DS3 controller is in use. is bluetooth audio disabled? thank u

**Reply by ndeadly** [reply]
Nothing regarding bluetooth audio has been touched. The only difference (from my side at least) between the DS3 and any other controller I support is the pairing procedure involving USB. The rest is mostly the Broadcom Bluetooth stack and some Nintendo code doing the work. I will take a look tomorrow morning and see if I can reproduce it on my console.

**Reply by ndeadly** [reply]
I can't reproduce this issue. Bluetooth audio continues to work with my Bose QC35 II headphones after connecting my DS3. There is a little noticable extra delay on the audio when using the controller, but I think this is normal for the Switch due to bandwidth limitations.

**Reply by DX12development** [reply]
> I can't reproduce this issue. Bluetooth audio continues to work with my Bose QC35 II headphones after connecting my DS3. There is a little noticable extra delay on the audio when using the controller, but I think this is normal for the Switch due to bandwidth limitations.

I think it is related to bandwidth cause I can use one airpod in mono and my DS3 controller at the same time, though the audio stutters a lot. But if I connect with both airpods it cuts out completely. Audio is fine if I'm just using joycons

**Reply by ndeadly** [reply]
This does sound like it could be bandwidth related. The Switch is severely limited in that regard and the experience is pretty mediocre even with official controllers. I think it was mostly intended for handheld play.

**Comment by nznznz746** [comment]
when Xbox2020 can be used  Author.respect

**Reply by ndeadly** [reply]
In the future.

**Reply by nznznz746** [reply]
omg

**Reply by nznznz746** [reply]
Will it have an approximate time? If it's too long, I think I should change the handle,

**Reply by ndeadly** [reply]
I don't give ETAs. Some time before the end of the year. Maybe.

**Reply by nznznz746** [reply]
i know, thanks，，

**Reply by nznznz746** [reply]
I tried for a long time and my switch couldn't search my PS5 controller Bluetooth

**Reply by nznznz746** [reply]
I don't know where the problem is

**Reply by ndeadly** [reply]
Have you read the readme?
https://github.com/ndeadly/MissionControl#usage
https://github.com/ndeadly/MissionControl#pairing-controllers

**Reply by ndeadly** [reply]
If you're following that and it's still not working, some things to try:
- make sure airplane mode isn't on (and if it is, you need to enable bluetooth separately)
- reset controller (there's a little button underneath that needs to be pushed with a pin or similar)
- fix archive bits in hekate
- ensure controller battery has enough charge
- delete `/config/MissionControl/controllers` directory

**Reply by nznznz746** [reply]
Does it need to be installed        Source code(zip)

**Reply by ndeadly** [reply]
No, only https://github.com/ndeadly/MissionControl/releases/download/v0.9.2/MissionControl-0.9.2-master-fe31665.zip

**Reply by nznznz746** [reply]
Sorry，it‘s still not work,I downloaded it, I installed it, but I just can't search for Bluetooth for PS5

**Comment by Pf-16** [comment]
Is it safe to update my dualsense to the latest version or will it not work with the latest release of mission control?

**Reply by ndeadly** [reply]
Yes. Updating controller firmware is generally fine, it's only Xbox One controllers that have been converted over to using Bluetooth LE that are problematic.

**Reply by Pf-16** [reply]
Ok, thanks

**Comment by oceanc220** [comment]
Can the wiiremote be recognized as a joycon so that I can play Mario Party with it?

**Reply by ndeadly** [reply]
No, not right now. It's possible to do it, but currently there are issues with motion controls I haven't been able to solve

**Reply by oceanc220** [reply]
Thank you, I hope you can solve this difficult problem, I am looking forward to it.

**Comment by Hcab8** [comment]
can i use a usb cable to connect dualsense controller

**Reply by ndeadly** [reply]
USB isn't currently supported. You could try sys-con for USB controllers

**Reply by Hcab8** [reply]
sys-con doesnt work either on 16.0.3. 😢

**Reply by ndeadly** [reply]
Try disabling Mission Control. It installs a USB event filter for Sony controllers for DS3 pairing. I haven't tested this but it may interfere with sys-con's detection of Sony controllers. It could also just be that Dualsense just isn't supported, I'm not sure.

**Reply by Hcab8** [reply]
failed again. it seems syscon just dont support dualsense. thanks though. 
i turn to usb because i want to have a even lower input lag. mission control works perfectily fine with bluetooth,hapic feedback is great,the progress is convenient. except that lags caused by bluetooth(maybe?) can still be felt, especially the lag of motion control. but its acceptable.👍

**Reply by ndeadly** [reply]
Supposedly the USB input path has greater lag than Bluetooth on the Switch. Never tested it myself but have seen it stated more than once.

**Reply by Masamune3210** [reply]
It does, at least ive seen a lot of people say it does. I have no easy way of testing it myself as my switch lives permanently in its dock

**Comment by unbreakable-7** [comment]
hi even though i have the latest atmosphere software my ps5 controller wont connect i tried lots of things but none of them worked what could be the problem

**Reply by ndeadly** [reply]
See https://github.com/ndeadly/MissionControl/discussions/618#discussioncomment-6195417 and the following post

**Reply by unbreakable-7** [reply]
i've already done all of those and it didnt work

**Reply by ndeadly** [reply]
Come join my [discord server](https://discord.gg/gegfNZ5Ucz). It's easier to provide support on these things there

**Reply by unbreakable-7** [reply]
ok

**Comment by Silicium3301** [comment]
Hi @ndeadly !
After installing the latest version of Mission Control, my Joy Cons began to be displayed in yellow and blue colors. Is there any way to fix this?

**Reply by ndeadly** [reply]
https://github.com/ndeadly/MissionControl/issues/621

**Comment by impeeza** [comment]
yeah, change the joy-cons case!   he he he.

testing now the latest version on my console and my joy-cons are showing the correct colors.

you can try All-in-One Switch Updater (AIO), on Tools, there is a item Change the Joy-Cons color.  you can backup your current settings and put another set of colors.

I just try to backup my config, put a new set of colors  then remove and connect two times the joy-cons to console and new color is displayed correctly. even if I have Mission control running.

**Comment by Hollusta1** [comment]
You are an absolute legend. Ive been leeching off github for years. I created an account just to tell you I am grateful for the time you put into this program. Thanks!

**Comment by wazoskyyy** [comment]
Hi i installed mission control on my switch, and it works, but the next day when i start my switch, and trying to open the CFW menu, it pops "fatal error occured when running atmosphere"

Can anyone help? Thanks a lot

**Reply by ndeadly** [reply]
Can't help you without an error report. Could literally be anything

**Reply by wazoskyyy** [reply]
Hi thanks, i figure it out, i deleted the file from mission control, and everything back to normal, my current firmware is 17.0.1 AMS 1.6.2
